src/olympia/hero/tests/test_models.py

```diff
@@ -68,25 +71,24 @@ def test_clean_gradient_and_image(self):
         with self.assertRaises(ValidationError) as ve:
             ph.clean()
         assert 'gradient_color' in ve.exception.error_dict
-        assert 'image' not in ve.exception.error_dict
+        assert 'select_image' not in ve.exception.error_dict
 
-        ph.update(image='foo.png')
+        ph.update(select_image=self.phi)
         with self.assertRaises(ValidationError) as ve:
             ph.clean()
         assert 'gradient_color' in ve.exception.error_dict
-        assert 'image' not in ve.exception.error_dict
+        assert 'select_image' not in ve.exception.error_dict
 
-        ph.update(image='', gradient_color='#123456')
+        ph.update(select_image=None, gradient_color='#123456')
         ph.clean()  # it raises if there's an error
 
     def test_clean_only_enabled(self):
         hero = PrimaryHero.objects.create(
-            promoted_addon=PromotedAddon.objects.create(
-                addon=addon_factory(), group_id=RECOMMENDED.id),
-            gradient_color='#C60184', image='foo.png')
-        PromotedApproval.objects.create(
-            version=hero.promoted_addon.addon.current_version,
-            group_id=RECOMMENDED.id)
+            disco_addon=DiscoveryItem.objects.create(addon=addon_factory()),
+            gradient_color='#C60184', select_image=self.phi)
+        hero.disco_addon.update(recommendable=True)
+        hero.disco_addon.addon.current_version.update(
+            recommendation_approved=True)
```

Comment:
and here - recommended status is now derived from PromotedAddon and PromotedApproval